Pod/Classes/PINURLSessionManager.m

```diff
@@ -102,7 +102,8 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didComp
     [self lock];
         dispatch_queue_t delegateQueue = self.delegateQueues[@(task.taskIdentifier)];
     [self unlock];
-    if ([(NSHTTPURLResponse *)task.response statusCode] == 404 && !error) {
+    NSHTTPURLResponse *response = (NSHTTPURLResponse *)task.response;
+    if ([response respondsToSelector:@selector(statusCode)] && [response statusCode] == 404 && !error) {
```

Comment:
This is a little odd because you've already cast the response to `NSHTTPURLResponse`, effectively asserting that it should have `statusCode`.

My suggestion:

```objc
if (!error && [task.response respondsToSelector:@selector(statusCode)] && [task.response statusCode] == 404) {
```